From de448507343a86ee97949eb4be0092d1659092da Mon Sep 17 00:00:00 2001
From: Adam Sampson <ats@offog.org>
Date: Sat, 18 Jul 2020 17:02:32 +0100
Subject: [PATCH 1/2] Remove unused variable.

This causes a duplicate symbol error with GCC 10 (-fno-common by
default), but it's not used anyway.
---
 src/playlist.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/src/playlist.c b/src/playlist.c
index 35483d1..96fac75 100644
--- a/src/playlist.c
+++ b/src/playlist.c
@@ -134,8 +134,6 @@ GtkWidget * plist__send_songs_to_iriver;
 GtkWidget * plist__export;
 #endif /* HAVE_TRANSCODING */
 
-gchar command[RB_CONTROL_SIZE];
-
 GtkTreeIter * fileinfo_iter = NULL;
 
 int playlist_dirty;

From 7f5d1f266957b3fa73799d3edef6b19ff9716d02 Mon Sep 17 00:00:00 2001
From: Adam Sampson <ats@offog.org>
Date: Sat, 18 Jul 2020 17:06:41 +0100
Subject: [PATCH 2/2] Rename timeout_tag to ports_timeout_tag.

This causes a duplicate symbol error with GCC 10 (-fno-common by
default), because there's also a global timeout_tag in main.c.

Also correct the type to guint rather than gint.
---
 src/ports.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/ports.c b/src/ports.c
index f17a5ab..f4de59d 100644
--- a/src/ports.c
+++ b/src/ports.c
@@ -63,7 +63,7 @@ GtkTreeViewColumn * column_out_R;
 int n_clients;
 GtkListStore * store_out_nb[MAX_JACK_CLIENTS];
 
-gint timeout_tag;
+guint ports_timeout_tag;
 
 int out_selector = 0;
 
@@ -242,7 +242,7 @@ tree_out_L_selection_changed(GtkTreeSelection * selection, gpointer * data) {
 			fprintf(stderr, "ERROR: jack_disconnect() returned %d\n", res);
 		}
 		g_free(str);
-		timeout_tag = aqualung_timeout_add(100, ports_timeout_callback, GINT_TO_POINTER(1));
+		ports_timeout_tag = aqualung_timeout_add(100, ports_timeout_callback, GINT_TO_POINTER(1));
         }
 }
 
@@ -262,7 +262,7 @@ tree_out_R_selection_changed(GtkTreeSelection *selection, gpointer * data) {
 			fprintf(stderr, "ERROR: jack_disconnect() returned %d\n", res);
 		}
 		g_free(str);
-		timeout_tag = aqualung_timeout_add(100, ports_timeout_callback, (gpointer)2);
+		ports_timeout_tag = aqualung_timeout_add(100, ports_timeout_callback, (gpointer)2);
         }
 }
 
